python/lsst/pipe/base/pipelineBuilder.py

```diff
@@ -195,97 +150,86 @@ def _newTask(self, pipeline, taskName, label=None, obsPkg=None, instrument=None)
 
             overrides.applyTo(config)
 
-        pipeline.append(TaskDef(taskName=taskName, config=config,
-                                taskClass=taskClass, label=label))
+        self._pipeline.append(TaskDef(taskName=taskName, config=config,
+                                      taskClass=taskClass, label=label))
 
-    def _dropTask(self, pipeline, label):
+    def deleteTask(self, label):
         """Remove task from a pipeline.
 
         Parameters
         ----------
-        pipeline : py:class:`Pipeline`
-            Pipeline instance.
         label : `str`
             Label of the task to remove.
         """
-        idx = pipeline.labelIndex(label)
+        idx = self._pipeline.labelIndex(label)
         if idx < 0:
             raise LookupError("Task label is not found: " + label)
-        del pipeline[idx]
+        del self._pipeline[idx]
 
-    def _moveTask(self, pipeline, label, new_idx):
+    def moveTask(self, label, new_idx):
```

Comment:
Is the index of a `PipelineTask` in a `Pipeline` or `PipelineBuilder` really user-visible?  I'd think it'd be better for these objects to make the ordering deterministic (i.e. topologically sorted given DatasetType input/output relationships, maybe lexicographical to break ties).  I'm not opposed to letting them having an unsorted state with an explicit validation or sort operation, but I don't think it's a good idea for them to expose an index-based interface to the Tasks.

Reply:
The index is not visible (though we can make it visible when doing `pipetask --show=pipeline`), but tasks are ordered in a pipeline as Pipeline is just a `list`. For execution we'll sort things topologically of course, but my idea was that people may prefer some tasks to be prioritized compared to others (within topological constraints) and one way to do it is re-ordering pipeline. If re-ordering is not useful I can certainly drop it, and there is always an option for an activator to implement its own prioritization later.